lib/wab/impl/data.rb

```diff
@@ -461,9 +461,9 @@ def detect_elememt(collection, key)
 
       def self.detect_string(s)
         len = s.length
-        if 36 == len && !/^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/.match(s).nil?
+        if 36 == len && WAB::Utils.uuid_format?(s)
```

Comment:
Why a function call instead of a const Regex? If a function, why not either pass in the length or check it instead of leaving half the detection outside the function?

Reply:
A const Regex is good. I wasn't sure if the regex above is created each time the function is called..


> why not either pass in the length or check it instead

I had considered that..
this particular function is used in a slightly different way in `WAB::Uuid`..

in `Controller` and `Data`, the length of the string is already cached in a variable. But not in `Uuid`. So this was the best way to replace with minimal collateral interference..

Reply:
I'm not sure it is created each time or not but it would make it easier to reuse if a const were used. Similar in that respect to a function but more efficient.

As for the length. If it is an argument to the function then it could be passed in or calculated and passed in. I'm not sure how expensive str.length is though. I'd hope it was cheap and could be called in a function without any penalty.

Reply:
How does the following sound?
- define the regex as constant(s) in the `Utils` module **and** calculate the string length in the util function..?

Reply:
I would keep the constants in the same file as the functions but otherwise, sure.

Reply:
since this PR is also about reducing duplicates, I'll define the constants in `Utils`..

Reply:
Since the functions are there that is perfect.